drivers/clk/clk-gpio.c

```diff
@@ -44,9 +45,37 @@
 struct clk_gpio {
 	struct clk_hw	hw;
 	struct gpio_desc *gpiod;
+	struct device *dev;
 };
 
 #define to_clk_gpio(_hw) container_of(_hw, struct clk_gpio, hw)
+static int clk_gpio_gate_acquire(struct clk_hw *hw)
+{
+	struct clk_gpio *clk = to_clk_gpio(hw);
+	struct device *dev = clk->dev;
+
+	clk->gpiod = devm_gpiod_get(dev, "enable", GPIOD_OUT_LOW);
+	if (IS_ERR(clk->gpiod))
+		return PTR_ERR(clk->gpiod);
+
+	return 0;
+}
+
+static int clk_gpio_gate_is_acquired(struct clk_hw *hw)
+{
+	struct clk_gpio *clk = to_clk_gpio(hw);
+
+	return clk->gpiod;
```

Comment:
`gpiod` is a pointer, not an `int`. Returning it as an int is therefore a weird cast (potentially truncating).

I'd be tempted to return a `bool` rather than `int`.